### PR TITLE
Update the query params according to the the filter changes

### LINKED ISSUE #152: Update the query params according to the the filter changes so the search result is shareable through the URL

### PR DESCRIPTION
**Pull request by ekamuktia**
Closes #152 

## Description

Update the query params according to the the filter changes so the search result is shareable through the URL

## Current Tasks
- [x] Initiate search based on url parameter
- [x] Change url parameter after search
- [x] Change autosearch using debounce

### CODE REVIEW

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Somehow the `Urut berdasarkan` dropdown doesn't select anything on the first page load.

Could you please solve this issue, Mbak @ekamuktia?

Screenshot 2021-07-20 at 11 33 24

**Comment by ekamuktia** [comment]
> Somehow the `Urut berdasarkan` dropdown doesn't select anything on the first page load.
> 
> Could you please solve this issue, Mbak @ekamuktia?
> 
> Screenshot 2021-07-20 at 11 33 24

Okay, I'll check

File: components/search-form.tsx [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { ChangeEvent, useState } from "react";
+import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
 
 import { PrimaryButton, SecondaryButton } from "./ui/button";
 import { Select } from "./ui/select";
 
+import { debounce } from "lodash";
+
```
Comment (mazipan) [inline components/search-form.tsx:11]: Hmmm, `lodash` 😴

Can we just create our own debounce? It's not that hard.
Once we add lodash, it's easy to abuse the usage

File: components/search-form.tsx [hunk 2]
```diff
@@ -26,36 +28,54 @@ export function SearchForm({
...
-  const [sort_by, setSortBy] = useState<string>("");
+  const [sort_by, setSortBy] = useState<string>(defaultSort);
 
   function handleSubmit(event: React.FormEvent<UsernameFormElement>) {
     event.preventDefault();
     onSubmitKeywords(keywords, filters, sort_by);
   }
 
-  function handleReset() {
+  function handleReset(event: React.FormEvent<UsernameFormElement>) {
```
Comment (mazipan) [inline components/search-form.tsx:54]: We already import the `ChangeEvent`, can we also move the `FormEvent` to the top level import?

File: components/search-form.tsx [hunk 3]
```diff
@@ -26,36 +28,54 @@ export function SearchForm({
...
+    query?: string;
+    filters?: {};
+    sort?: string;
+  };
 }) {
+  const defaultSort = sortSettings?.length ? sortSettings[0].value : "";
   const [keywords, setKeywords] = useState<string>("");
   const [filters, setFilters] = useState<any>({});
-  const [sort_by, setSortBy] = useState<string>("");
+  const [sort_by, setSortBy] = useState<string>(defaultSort);
```
[the remark this answers is not in the document]
Comment (ekamuktia) [reply components/search-form.tsx]: Okay, it's fixed now

File: components/search-form.tsx (hunk 1 above)
Comment (ekamuktia) [reply components/search-form.tsx:17]: I changed it to use `ts-debounce` instead, could you check if it's okay? Thank you

**Review by mazipan — APPROVED** [review]
Look 🐐 for me.

Ship it!

File: components/search-form.tsx (hunk 1 above)
Comment (mazipan) [reply components/search-form.tsx:17]: https://bundlephobia.com/package/ts-debounce@3.0.0

**Review by zainfathoni — APPROVED** [review]
LGTM 💯 

Thanks for working on it, Mbak @ekamuktia! 🙏

**Pull request merged**